pythonAPIClient/client.py

```diff
@@ -160,48 +159,70 @@ def search_camera(self, latitude=None, longitude=None, radius=None, camera_type=
+        country : str, optional
+        state : str, optional
+        city : str, optional
+        resolution_width : int, optional
+        resolution_height : int, optional
+        is_active_image : bool, optional
+        is_active_video : bool, optional
+
+        Raises
+        ------
+        FormatError
+            If type of argument value is not expected for the given field.
+
+            Or radius cannot is less than 0.
+
+            Or incorrect latitude range. (it should be between +90 and -90)
+
+            Or incorrect longitude range. (it should be between +180 and -180)
+
+        AuthenticationError
+            If the client secret of this client object does not match the clientID.
+        InternalError
+            If there is an API internal error.
+
+        """
         if self.token is None:
             self.request_token()
-        url = Client.base_URL + 'cameras/search?'
-        if latitude is not None:
-            url += 'lat=' + latitude + '&'
-        if longitude is not None:
-            url += 'lng=' + longitude + '&'
-        if radius is not None:
-            url += 'radius=' + radius + '&'
-        if camera_type is not None:
-            url += 'type=' + camera_type + '&'
-        if source is not None:
-            url += 'source=' + source + '&'
-        if country is not None:
-            url += 'country=' + country + '&'
-        if state is not None:
-            url += 'state=' + state + '&'
-        if city is not None:
-            url += 'city=' + city + '&'
-        if resolution_width is not None:
-            url += 'resolution_width=' + resolution_width + '&'
-        if resolution_heigth is not None:
-            url += 'resolution_heigth=' + resolution_heigth + '&'
-        if is_active_image is not None:
-            url += 'is_active_image=' + is_active_image + '&'
-        if is_active_video is not None:
-            url += 'is_active_video=' + is_active_video + '&'
-        if offset is not None:
-            url += 'offset=' + offset + '&'
-        url = url[:-1]
-        response = requests.get(url, headers=self.header_builder())
+        local_params = dict(locals())
+        local_params.pop('self', None)
+        local_params['type'] = local_params.pop('camera_type', None)
+
+        # filter out those parameters with value None, change true/false
+        search_params = {k: v for k, v in local_params.items() if v is not None}
+
+        url = Client.base_URL + 'cameras/search'
+
+        response = requests.get(url, headers=self.header_builder(), params=search_params)
         if response.status_code == 401:
             self.request_token()
-            response = requests.get(url, headers=self.header_builder())
+            response = requests.get(url, headers=self.header_builder(), params=search_params)
         elif response.status_code == 422:
             raise FormatError(response.json()['message'])
         elif response.status_code == 500:
```

Comment:
Since you are already checking `elif status code != 200:` , I don't think you need to check 500 anymore right